### PR TITLE
chore: improve `dfx start` messages.

### PR DESCRIPTION
**Pull request by vincent-dfinity**
# Description

1. Improve `dfx start` messages to let users know what to do next clearly.
2. Add `PostStart` actor which will be notified by `PocketIcProxy` actor to output the messages in the right order.
3. Set the `original_background` environment variable which will be used by the second start, as for now with `--background` we will run the `dfx start` again without `--background` option, but we need its value in PostStart actor.

Fixes # (issue)

[SDK-1844](https://dfinity.atlassian.net/browse/SDK-1844)

# How Has This Been Tested?

Manually tests.
```
$ dfx start
Running dfx start for version 0.23.0-beta.3+rev34.dirty-11793309
Using the default configuration for the local shared network.
Initialized replica.
Initialized HTTP gateway.
Replica API running on 127.0.0.1:4943
The dfx server is running.
You can start a new terminal to continue developing, or quit with 'Ctrl-C'.
```
```
$ dfx start --background
Running dfx start for version 0.23.0-beta.3+rev34.dirty-11793309
Using the default configuration for the local shared network.
Initialized replica.
Initialized HTTP gateway.
Replica API running on 127.0.0.1:4943
The dfx server is running in the background.
```

Also add e2e test for `dfx start --background`, not sure how to add test for `dfx start`...

# Checklist:

- [x] The title of this PR complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I have edited the CHANGELOG accordingly.
- [ ] I have made corresponding changes to the documentation.

[SDK-1844]: https://dfinity.atlassian.net/browse/SDK-1844?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by marydwyer** [comment]
Because new user often don't realize that they successfully ran dfx start and need to open a new terminal, I recommend making it even more clear.

For dfx start:
`✅ Success! The dfx server is running.`
`➕ You must open a new terminal to continue developing. If you'd prefer to stop, quit with 'Ctrl-C'.`

For dfx start --background:
`✅ Success! The dfx server is running in the background.`

File: src/dfx/src/commands/start.rs [hunk 1]
```diff
@@ -264,6 +265,10 @@ pub fn exec(
     }
     local_server_descriptor.describe(env.get_logger());
 
+    // Get the original background flag set by the user from the command arguments.
+    // Get it from the environment variable as the `--background` flag will be ignored by the send_background() method.
+    let original_background = std::env::var("original_background").is_ok();
```
Comment (ericswanson-dfinity) [inline src/dfx/src/commands/start.rs:270]: We can define this in the clap definition, but still use an environment variable:

```
#[arg(env = "DFX_RUNNING_IN_BACKGROUND", hide = true)]
bool running_in_background;
```

This way we're declaring it with the rest of the CLI inputs, but it still won't show up in help or affect the shell completion scripts.

File: src/dfx/src/actors/post_start.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+impl Handler<PocketIcProxyReadySignal> for PostStart {
+    type Result = ();
+
+    fn handle(&mut self, _msg: PocketIcProxyReadySignal, _ctx: &mut Self::Context) -> Self::Result {
+        let logger = &self.config.logger;
+        if self.config.background {
+            info!(logger, "The dfx server is running in the background.")
+        } else {
+            info!(logger, "The dfx server is running.\nYou can start a new terminal to continue developing, or quit with 'Ctrl-C'.");
```
Comment (ericswanson-dfinity) [inline src/dfx/src/actors/post_start.rs:53]: I agree with Mary's suggestions for these messages

File: src/dfx/src/commands/start.rs (hunk 1 above)
Comment (vincent-dfinity) [reply src/dfx/src/commands/start.rs]: Fixed at https://github.com/dfinity/sdk/pull/3946/commits/060b25638ae42ecbf2c5854962084a6c5851781e

File: src/dfx/src/actors/post_start.rs (hunk 2 above)
Comment (vincent-dfinity) [reply src/dfx/src/actors/post_start.rs]: Fixed at https://github.com/dfinity/sdk/pull/3946/commits/060b25638ae42ecbf2c5854962084a6c5851781e

**Review by ericswanson-dfinity — DISMISSED** [review]

**Review by ericswanson-dfinity — APPROVED** [review]

**Pull request merged**